### PR TITLE
chore(rpc-types-compat): use SealedBlock.raw_transactions to reduce deplicate

### PR DESCRIPTION
**Pull request by jsvisa**
Seems the `SealedBlock.body` is used in the v1,v2,v3 payload builder, use a helper function to reduce the same logic.

### CODE REVIEW

File: crates/primitives/src/block.rs [hunk 1]
```diff
@@ -348,6 +348,18 @@ impl SealedBlock {
 
         Ok(())
     }
+
+    /// Returns a vector of raw formated transactions.
```
Comment (Rjected) [inline crates/primitives/src/block.rs:352]:
```suggestion
    /// Returns a vector of transactions RLP encoded with [TransactionSigned::encode_enveloped].
```

File: crates/primitives/src/block.rs [hunk 2]
```diff
@@ -348,6 +348,18 @@ impl SealedBlock {
...
+    pub fn raw_transactions(&self) -> Vec<Bytes> {
+        self.body
+            .iter()
+            .map(|tx| {
+                let mut encoded = Vec::new();
+                tx.encode_enveloped(&mut encoded);
+                encoded.into()
```
Comment (Rjected) [inline crates/primitives/src/block.rs:359]: can use `envelope_encoded` here

**Review by Rjected — DISMISSED** [review]
just a few style nits, otherwise I like removing the code dup

**Comment by Rjected** [comment]
this also needs a rebase to fix the test failures

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**